workspaces/rbac/plugins/rbac-backend/migrations/20250305155143_migration.js

```diff
@@ -15,12 +15,48 @@
  */
 
 exports.up = async function up(knex) {
-  const isRoleMetaDataExist = await knex.schema.hasTable('role-metadata');
-  if (isRoleMetaDataExist) {
+  const roleMetaDataExist = await knex.schema.hasTable('role-metadata');
+  const casbinExists = await knex.schema.hasTable('casbin_rule');
+  if (roleMetaDataExist) {
+    // Add the owner field to the role-metadata field
     await knex.schema.alterTable('role-metadata', table => {
       table.string('owner');
     });
   }
+
+  if (casbinExists && roleMetaDataExist) {
+    // Get the policies for resource type policy-entity and action create
+    const policyEntityCreateRoles = await knex
+      .from('casbin_rule')
+      .where('v1', 'policy-entity')
+      .where('v2', 'create')
+      .pluck('v0');
+
+    // Ensure that we are only updating the rest api and configuration policies only
+    const rolesFromConfigAndRest = await knex
+      .from('role-metadata')
+      .whereNot('source', 'csv-file')
+      .whereIn('roleEntityRef', policyEntityCreateRoles)
+      .pluck('roleEntityRef');
+
+    // Update the polices from the config and rest from resource type policy-entity and action create
+    // to policy.entity.create and action create
+    await knex
+      .from('casbin_rule')
+      .whereIn('v0', rolesFromConfigAndRest)
+      .where('v2', 'create')
+      .update({ v1: 'policy.entity.create' });
+
+    const rolesFromCSV = await knex
+      .from('role-metadata')
+      .where('source', 'csv-file')
+      .whereIn('roleEntityRef', policyEntityCreateRoles)
+      .pluck('roleEntityRef');
+
+    console.log(
```

Comment:
I think console.warn will be better here

Reply:
It doesn't seem like `console.warn` works in the migration file. We will have to leave it as `console.log`